src/rpc/client.cpp

```diff
@@ -171,3 +191,33 @@ UniValue RPCConvertValues(const std::string &strMethod, const std::vector<std::s
 
     return params;
 }
+
+UniValue RPCConvertNamedValues(const std::string &strMethod, const std::vector<std::string> &strParams)
+{
+    UniValue params(UniValue::VOBJ);
+
+    for (const std::string &s: strParams) {
+        unsigned ptr = 0;
+        if (s.size() <= ptr || s[ptr] != '-') {
+            throw(std::runtime_error("Non-named or empty argument found"));
+        }
+        ptr += 1;
+        if (s.size() > ptr && s[ptr] == '-') // Allow "--opt" as well as "-opt"
+            ptr += 1;
+        size_t pos = s.find("=", ptr);
+        if (pos == std::string::npos) {
+            throw(std::runtime_error("No '=' in named argument, a value needs to be provided for every argument"));
+        }
+        std::string name = s.substr(ptr, pos-ptr);
+        std::string value = s.substr(pos+1);
```

Comment:
In that case it should do the expected thing: send an empty name or value.
An empty value is perfectly acceptable if you want to send an empty string.
An empty name is a bit strange, agreed, but it gets caught server-side as an unknown argument.